### PR TITLE
Readability and efficiency commit

### PR DESCRIPTION
**Pull request by iamcharleschege**
This commit improves the readability of the code and increases efficiency by:
1. Improving readability of HEX values by adding _ to remove warnings and getting rid of redundant fields in favour of shorthand values
2. Removing static lifetime value to constants as  by default they have static lifetimes 
3. Passing by value instead of by reference to improve efficiency in some areas

### CODE REVIEW

**Comment by breeswish** [comment]
Great! Seems that many of them can be checked by clippy. @BusyJay

File: src/async/executor.rs [hunk 1]
```diff
@@ -182,10 +183,10 @@ impl<'a> Executor<'a> {
     /// pair by yourself.
     pub fn spawn<F>(&self, f: F)
     where
-        F: Future<Item = (), Error = ()> + Send + 'static,
+    F: Future<Item = (), Error = ()> + Send + 'static,
```
[the remark this answers is not in the document]
Comment (iamcharleschege) [reply src/async/executor.rs]: Changes were made with the help of clippy. By introducing reference, unnecessary copying was eliminated.

File: src/async/executor.rs [hunk 2]
```diff
@@ -182,10 +183,10 @@ impl<'a> Executor<'a> {
...
     {
         let s = executor::spawn(Box::new(f) as BoxFuture<_, _>);
         let notify = Arc::new(SpawnNotify::new(s, self.cq.clone()));
-        poll(notify, false)
+        poll(&notify, false)
     }
-}
+}
```
[the remark this answers is not in the document]
Comment (iamcharleschege) [reply src/async/executor.rs]: I don't understand what you mean on this commnet. Please explain further

[the remark this answers is not in the document]
Comment (Hoverbear) [reply src/async/executor.rs]: It's customary with git to have files end with a newline. Here your file ends with `}`.

File: src/async/promise.rs [hunk 3]
```diff
@@ -60,7 +60,7 @@ impl Batch {
                 guard.set_result(Ok(None))
             }
         };
-        task.map(|t| t.notify());
+        if let Some(t) = task { t.notify() };
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply src/async/promise.rs]: This is a clippy suggestion if I recall.

File: src/async/callback.rs [hunk 4]
```diff
@@ -24,8 +24,8 @@ pub struct Request {
 
 impl Request {
     pub fn new(inner: Arc<ServerInner>) -> Request {
-        let ctx = RequestContext::new(inner);
-        Request { ctx: ctx }
+        let new_ctx = RequestContext::new(inner);
+        Request { ctx: new_ctx }
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply src/async/callback.rs]: Or just using:

```
let ctx = RequestContext::new(inner);
Request { ctx }
```

File: src/async/executor.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (BusyJay) [reply src/async/executor.rs]: But the change is just removing an indent.

File: src/async/promise.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (BusyJay) [reply src/async/promise.rs]: Why prefer `if let` than `map` then? I think it's more natural for a functional programming language to prefer `map` than condition check.

File: src/async/callback.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (iamcharleschege) [reply src/async/callback.rs]: Yeah. It was a mistake but I have already made the changes. Making a few more changes then I will PR

File: src/async/executor.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (iamcharleschege) [reply src/async/executor.rs]: Settings to my editor automatically remove unnecessary indentation by default. Nothing major here

File: src/async/promise.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (iamcharleschege) [reply src/async/promise.rs]: We can just use functional programming. No problem. The suggestion was from clippy

**Comment by BusyJay** [comment]
CI fails.

**Comment by siddontang** [comment]
cool @iamcharleschege 

can you fix the CI?

**Comment by iamcharleschege** [comment]
@siddontang I already did and the results were successful and were merged

**Comment by BusyJay** [comment]
Close as #207 has already been merged.

**Pull request closed without merging**